cmd/server-mux.go

```diff
@@ -194,6 +199,9 @@ func newListenerMux(listener net.Listener, config *tls.Config) *ListenerMux {
 				return
 			}
 
+			// Enable Read timeout
+			conn.SetReadDeadline(time.Now().Add(defaultTcpReadTimeout))
+
 			// Enable keep alive for each connection.
```

Comment:
> Tested this thoroughly with some other changes the problem is that once this happens for example
> 
> 2017/01/22 11:52:22 http: Accept error: accept tcp [::]:9000: accept4: too many open files; retrying in 5ms
> Golang stops responding especially "AccepTCP()" is hung for ever.

This change doesn't avoid "too many connections" error, it just evicts silent clients, that are silent for a long time (we can define what this long time is)